src/metalperformanceshaders.cs

```diff
@@ -4203,159 +6066,2016 @@ interface MPSImageReduceUnary {
+		IntPtr Constructor (NSCoder decoder, IMTLDevice device);
+
+		[Export ("initWithDevice:keepProbability:seed:maskStrideInPixels:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float keepProbability, nuint seed, MTLSize maskStrideInPixels);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnGradientKernel), Name = "MPSCNNDropoutGradient")]
+	[DisableDefaultCtor]
+	interface MPSCnnDropoutGradient {
+
+		[Export ("keepProbability")]
+		float KeepProbability { get; }
+
+		[Export ("seed")]
+		nuint Seed { get; }
+
+		[Export ("maskStrideInPixels")]
+		MTLSize MaskStrideInPixels { get; }
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder decoder, IMTLDevice device);
+
+		[Export ("initWithDevice:keepProbability:seed:maskStrideInPixels:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float keepProbability, nuint seed, MTLSize maskStrideInPixels);
+	}
+
+	[TV (11, 3), Mac (10, 13, 4, onlyOn64: true), iOS (11, 3)]
+	[BaseType (typeof (MPSNNGradientState), Name = "MPSCNNInstanceNormalizationGradientState")]
+	[DisableDefaultCtor]
+	interface MPSCnnInstanceNormalizationGradientState {
+
+		[Export ("instanceNormalization", ArgumentSemantic.Retain)]
+		MPSCnnInstanceNormalization InstanceNormalization { get; }
+
+		[NullAllowed, Export ("gamma")]
+		IMTLBuffer Gamma { get; }
+
+		[NullAllowed, Export ("beta")]
+		IMTLBuffer Beta { get; }
+
+		[Export ("gradientForGamma")]
+		IMTLBuffer GradientForGamma { get; }
+
+		[Export ("gradientForBeta")]
+		IMTLBuffer GradientForBeta { get; }
+	}
+
+	interface IMPSCnnInstanceNormalizationDataSource { }
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject), Name = "MPSCNNInstanceNormalizationDataSource")]
+	interface MPSCnnInstanceNormalizationDataSource : NSCopying {
+
+		[Abstract]
+		[Export ("gamma")]
+		IntPtr /* float* */ Gamma { get; }
+
+		[Abstract]
+		[Export ("beta")]
+		IntPtr /* float* */ Beta { get; }
+
+		[Abstract]
+		[Export ("numberOfFeatureChannels")]
+		nuint NumberOfFeatureChannels { get; }
+
+		[Abstract]
+		[Export ("label")]
+		string Label { get; }
+
+		[Export ("updateGammaAndBetaWithCommandBuffer:instanceNormalizationStateBatch:")]
+		[return: NullAllowed]
+		MPSCnnNormalizationGammaAndBetaState UpdateGammaAndBeta (IMTLCommandBuffer commandBuffer, MPSCnnInstanceNormalizationGradientState [] instanceNormalizationStateBatch);
+
+		[Export ("updateGammaAndBetaWithInstanceNormalizationStateBatch:")]
+		bool UpdateGammaAndBeta (MPSCnnInstanceNormalizationGradientState [] instanceNormalizationStateBatch);
+
+		[Export ("epsilon")]
+		float GetEpsilon ();
+
+		[Export ("encodeWithCoder:")]
+		void Encode (NSCoder coder);
+
+		[Export ("initWithCoder:")]
+		IntPtr Constructor (NSCoder decoder);
+
+		[Static]
+		[Export ("supportsSecureCoding")]
+		bool SupportsSecureCoding { get; }
```

Comment:
`[Static]` here makes me think the binding won't work.
It's likely `static-per-type`, so if `Foo` conforms to the protocol then `Foo.SupportsSecureCoding` might be true, while `Bar`, also conforming, might be `false`

Reply:
Yes you are right, this needs to be exposed into implementors of the protocol, does not make much sense exposing it here. 👍